### PR TITLE
pylint, update docs and CI

### PR DESCRIPTION
**Pull request by jmborr**
### Work Done
- correct a few typos in the docs
- update contributing docs
- update CI so that it can be run on forked repos (as a test before submitting PR's to reflectometry/orsopy)

### CODE REVIEW

**Comment by jmborr** [comment]
@aglavic @arm61 
Hi, I started reading `orsopy` because I'll most likely end up using it to write reduced files in package [reflectivity_ui](https://github.com/neutrons/reflectivity_ui). 
While reading, I made some minor changes to the docs as well as the CI so that [I could run it in my forked repo](https://github.com/jmborr/orsopy/pull/1) before pestering the original repo with this PR :slightly_smiling_face:

File: CONTRIBUTING.rst [hunk 1]
```diff
@@ -97,15 +97,18 @@ Ready to contribute? Here's how to set up `orsopy` for local development.
...
+check that it meets these guidelines:
 
-1. The pull request should include tests.
 2. If the pull request adds functionality, the docs should be updated. Put
    your new functionality into a function with a docstring, and add the
    feature to the list in README.rst.
-3. The pull request should work for Python 3.5, 3.6, 3.7 and 3.8, and for PyPy. Check
-   https://travis-ci.com/reflectivity/orsopy/pull_requests
-   and make sure that the tests pass for all supported Python versions.
+1. The pull request should include tests for the new functionality. Run the tests in your local machine with `pytest`.
```
Comment (andyfaff) [inline CONTRIBUTING.rst:106]: This list won't be ordered now. 2 appears to come before 1.

File: CONTRIBUTING.rst [hunk 2]
```diff
@@ -97,15 +97,18 @@ Ready to contribute? Here's how to set up `orsopy` for local development.
...
+3. The pull request should work for Python 3.5, 3.6, 3.7 and 3.8, and for PyPy. To make sure that the tests pass
```
Comment (andyfaff) [inline CONTRIBUTING.rst:107]:
```suggestion
3. The pull request should work for Python 3.6, 3.7, 3.8, 3.9, 3.10, 3.11 and for PyPy. To make sure that the tests pass
```

3.6 and 3.7 are kind of supported, in that there's a desire for older Python versions to keep working if possible. 3.5 is too old. So `[3.6, 3.7]` if we can keep it working, but definitely `[3.8, 3.9, "3.10", "3.11", "3.12"]`.

**Review by andyfaff — COMMENTED** [review]
Looks good, apart from a few minor changes. Thank you for contributing. Sorry I missed this PR, it must have been caught up in notification noise in my inbox.

**Comment by aglavic** [comment]
Thanks for contributing, looks good to me. Will be included in the next release.

File: CONTRIBUTING.rst (hunk 1 above)
Comment (andyfaff) [reply CONTRIBUTING.rst:106]: @aglavic this list is unordered now, and the formatting is wrong. It would've been better to fix the PR as asked.

Comment (jmborr) [reply CONTRIBUTING.rst:106]: @andyfaff I have a mind of creating a conda recipe for orsopy so I will include fixes for all your comments in my next PR 😁

Comment (andyfaff) [reply CONTRIBUTING.rst:106]: There's a staged recipe on condaforge that needs the cf maintainers to do something

Comment (aglavic) [reply CONTRIBUTING.rst:106]: Sorry, I thought those would have been done already. My fault.

**Pull request merged**